components/net/http_loader.rs

```diff
@@ -1737,6 +1760,29 @@ impl Drop for ResponseEndTimer {
     }
 }
 
+fn prompt_user_for_credentials(
+    embedder_proxy: &Mutex<EmbedderProxy>,
+) -> Option<PromptCredentialsInput> {
+    let proxy = embedder_proxy.lock().unwrap();
+
+    let (ipc_sender, ipc_receiver) = ipc::channel().unwrap();
+
+    proxy.send((
+        None,
+        EmbedderMsg::Prompt(
+            PromptDefinition::Credentials(ipc_sender),
+            PromptOrigin::Trusted,
+        ),
+    ));
```

Comment:
This should contain the ID of the browsing context. Otherwise, the embedder has no idea where it comes from.

Reply:
Could you open an issue?